### PR TITLE
chore(rfcs): Kubernetes Metrics API

### PR DESCRIPTION
**Pull request by MOZGIII**
[Rendered](https://github.com/timberio/vector/blob/k8s-metrics-api-rfc/rfcs/2020-09-16-3875-kubernetes-metrics-api.md)

Ref: #3875 (not closing as there's more work to do)

### CODE REVIEW

**Comment by jamtur01** [comment]
@MOZGIII I am trying to understand the conclusion reached here. Are we closing this RFC in favor of just telling people to use the `prometheus` source?

File: rfcs/2020-09-16-3875-kubernetes-metrics-api.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+## Should we use it
+
+Overall, it looks like Metrics API is designed to allow external monitoring
+tools to route data in such a way that it's available in a "native" fashion
+to the Horizontal Pod Autoscaler and other Kubernetes components that rely on
+this data. This means that while we _can_ integrate with the Metrics API
+directly, it might not be the best way, and going with an alternative - like
+integrating with those external metrics aggregation system via the Prometheus
+scraping protocol- might be a better approach.
```
Comment (binarylogic) [inline rfcs/2020-09-16-3875-kubernetes-metrics-api.md:192]: Can you clarify what this means? I didn't see any previous mention of an external metrics aggregation system.

**Comment by binarylogic** [comment]
@MOZGIII thanks for writing this up. Given https://github.com/timberio/vector/pull/4098#issuecomment-699549393, let's close this. It appears that collecting metrics from K8s will be covered by #4162 and #4163.

File: rfcs/2020-09-16-3875-kubernetes-metrics-api.md (hunk 1 above)
Comment (MOZGIII) [reply rfcs/2020-09-16-3875-kubernetes-metrics-api.md:192]: I have a section on this at the metrics overview at https://github.com/timberio/vector/pull/4098

**Pull request closed without merging**